pkg/actions/lua/net/http/request.go

```diff
@@ -0,0 +1,133 @@
+package http
+
+import (
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/Shopify/go-lua"
+	"github.com/treeverse/lakefs/pkg/actions/lua/util"
+)
+
+const defaultRequestTimeout = 30 * time.Second
+
+func Open(l *lua.State) {
+	open := func(l *lua.State) int {
+		lua.NewLibrary(l, httpLibrary)
+		return 1
+	}
+	lua.Require(l, "net/http", open, false)
+	l.Pop(1)
+}
+
+var httpLibrary = []lua.RegistryFunction{
+	{Name: "request", Function: httpRequest},
+}
+
+// httpRequest - perform http request
+//
+//	Accepts arguments (url, body) or table with url, method, body, headers. Value for url is required.
+//		method is by default GET or POST in case body is set.
+//	Returns code, body, headers, status.
+func httpRequest(l *lua.State) int {
+	req := prepareRequest(l)
+	client := http.Client{
+		Timeout: defaultRequestTimeout,
+	}
+	resp, err := client.Do(req)
+	check(l, err)
+	defer func() { _ = resp.Body.Close() }()
+	body, err := io.ReadAll(resp.Body)
+	check(l, err)
+
+	// push return
+	l.PushInteger(resp.StatusCode)
+	l.PushString(string(body))
+	pushResponseHeader(l, resp.Header)
+	l.PushString(resp.Status)
+	return 4
+}
+
+func prepareRequest(l *lua.State) *http.Request {
+	var (
+		reqMethod  = http.MethodGet
+		reqURL     string
+		reqBody    io.Reader
+		reqHeaders map[string]interface{}
+	)
+	switch l.TypeOf(1) {
```

Comment:
Can we create internal consts to these indexes so that it will be more readable for us?

Reply:
it is common practice to use the arg position as a number and not as const for the lua func implementation

Reply:
I understand, but can we at least document what these numbers mean?